draft/0001-organization.md

```diff
@@ -0,0 +1,273 @@
+
+Organization permissions (scopes) are distinct from the `scope` in OpenID Connect and [Resource Indicators](https://www.rfc-editor.org/rfc/rfc8707.html#name-access-token-request). While these existing specifications share terminology like "scope" and "role" with organization, they are designed to address different authentication and authorization challenges. Organization scopes and roles should be treated separately.
+
+## 5. Authentication
+
+In OpenID Connect, the [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) is a JWT token containing user identity information. In addition to the [Standard Claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims), we introduce new claims to the ID token to support organization.
+
+### 5.1 Organization claims
+
+| Name               | Type                  | Description                                                                                                                                                                                          |
+| ------------------ | --------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| organization_ids   | array of strings      | The unique organization identifiers that the user has a membership with.                                                                                                                             |
+| organization_roles | array of JSON objects | An array of organization roles that the user has. Each object contains the following fields: <ul><li>`organization_id`: The unique organization identifier as a string.</li><li>`roles`: The role names granted to the user within the organization.</li></ul> |
+
+The objects in the `organization_roles` array must adhere to these rules:
+
+- The `organization_id` field must be unique within the array.
+- The `roles` field must be an array of strings, with each string being unique within the array.
+- The `roles` field must not be empty.
+
+These rules help minimize the size of the ID token and can be enforced by the OpenID Provider (OP) when issuing the ID token.
+
+An example of the ID token with organization claims:
+
+```json
+{
+  "sub": "12345",
+  "name": "John Wick",
+  "organization_ids": ["12345", "67890", "13579"],
+  "organization_roles": [
+    {
+      "organization_id": "12345",
+      "roles": ["admin"]
+    },
+    {
+      "organization_id": "67890",
+      "roles": ["viewer", "editor"]
+    }
+  ]
+}
+```
+
+### 5.2 Requesting claims using scope values
+
+As per the existing authentication flow, the OpenID Provider (OP) SHOULD accept an additional `scope` value in the authentication request and provide the corresponding claim in the ID token:
+
+- `organization`: This scope value requests access to the `organization_ids` and `organization_roles` claims.
```

Comment:
Since this is not a command standard, better give it a prefix?  It will become a reserved scope term by Logto.

Reply:
Want to confirm that, this is a special, or preserved scope for Logto Orginization?

Reply:
there's no recommendation for claim naming in [ID token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken), or [addtional claims](https://openid.net/specs/openid-connect-core-1_0.html#AdditionalClaims), specifically.

this reminds me the argument of appending `X-` in custom headers. i think it's ok to add them directly

Reply:
Claim naming is fine, how about the `organization` scope here?

Reply:
i'm good with a scoped scope name though

Reply:
updated